scripts/create_incr_tickets.py

```diff
@@ -0,0 +1,169 @@
+
+    def rebalance_root(self):
+        """
+        update the root of this batch after a file has been added, in case
+        their paths differ. For example:
+
+        if this batch had a root of /a/b/c and we add a file from /a/b/d,
+        the newly balanced root should be /a/b
+        """
+        split_dirs = map(lambda d: d.split('/'), self.directories)
+        new_root = []
+        for level in zip(*split_dirs):
+            if not(all(map(lambda d: d == level[0], level))):
+                break
+            new_root.append(level[0])
+        self.root = '/'.join(new_root)
+
+    def file_count(self):
+        return len(self.files)
+
+    def blocks(self, dirs):
+        """
+        determine if this batch of files blocks work on another batch of
+        files. This is the case when a subdir
+        """
+        return any(map(lambda d: d in self.directories, dirs))
+
+    def base_similar(self, other_root):
+        """
+        determine if this batch has a root that is similar to another- that is,
+        it is either the same, is a subdirectory, or they share a common parent
+        """
+        if self.root == other_root:
+            return True
+        elif self.root.split('/')[:-1] == other_root:
+            return True
+        elif self.root.split('/')[:-1] == other_root.split('/')[:-1]:
+            return True
+        elif other_root in self.root:
+            return True
+        else:
+            return False
+
+def check_if_blocked(batches, root, dirs):
+    """
+    djfdj
+    """
+    paths = [os.path.join(root, d) for d in dirs]
+    return any(map(lambda b: b.blocks(paths), batches))
+
+def filter_python_files(files):
+    """
+    given a list of files, extract the python files
+    """
+    return filter(lambda f: f.split('.')[-1] == 'py', files)
+
+def crawl(path, LIMIT):
+    """
+    crawl a given file path, from the deepest node up, collecting and
+    organizing directories containing python files into `Batches` of less
+    than `LIMIT` python files.
+    """
+    batches = []
+    in_a_batch = False
+
+    for root, dirs, files in os.walk(path, topdown=False):
+        # skip directories that have no python files
+        if len(filter_python_files(files)) < 1:
+            continue
+        if not in_a_batch:
+            current_batch = Batch(root)
+            in_a_batch = True
+        if not current_batch.base_similar(root):
+            batches.append(current_batch)
+            current_batch = Batch(root)
+            in_a_batch = True
+        # mark this batch as `blocked` if any of the subdirectories in the
+        # current node have already been added to the list of batches
+        if check_if_blocked(batches, root, dirs):
+            current_batch.blocked = True
+        python_files = [os.path.join(root, f) for f in filter_python_files(files)]
+        for file_path in python_files:
+            current_batch.add(file_path)
+        if current_batch.file_count() >= LIMIT:
```

Comment:
We're comparing against the limit after adding the files, so each batch will typically be a bit over the limit?

Reply:
Yeah. There is the case where we hit a very large group of python files and it exceeds the limit, but I have found that setting the limit to 10 and comparing the file limit after adding the current files results in relatively well balanced groups. At the very least, I could explain myself better, but what would you propose here? I suppose I can change this now that we know python-modernize can accept regex in file-paths